jdbc/src/main/java/nextstep/jdbc/JdbcTemplate.java

```diff
@@ -23,56 +23,56 @@ public JdbcTemplate(final DataSource dataSource) {
     }
 
     public <T> List<T> query(String sql, RowMapper<T> rowMapper, Object... parameters) {
-        try (
-            Connection conn = dataSource.getConnection();
-            PreparedStatement pstmt = conn.prepareStatement(sql)
-        ) {
-            bindParameters(pstmt, parameters);
-            ResultSet rs = pstmt.executeQuery();
+        Executor<List<T>> executor = (pstmt) -> {
+            try (ResultSet rs = pstmt.executeQuery()) {
+                List<T> result = new ArrayList<>();
+                int rowNum = 1;
 
-            log.debug("query : {}", sql);
-
-            List<T> result = new ArrayList<>();
-            int rowNum = 1;
+                while (rs.next()) {
+                    result.add(rowMapper.mapRow(rs, rowNum));
+                }
 
-            while (rs.next()) {
-                result.add(rowMapper.mapRow(rs, rowNum));
+                return result;
             }
+        };
 
-            rs.close();
-            return result;
-        } catch (SQLException e) {
-            log.error(e.getMessage(), e);
-            throw new RuntimeException(e);
-        }
+        return executeQuery(executor, sql, parameters);
     }
 
     public <T> T queryForObject(String sql, RowMapper<T> rowMapper, Object... parameters) {
         List<T> result = query(sql, rowMapper, parameters);
 
         if (result.isEmpty()) {
-            throw new RuntimeException("쿼리 결과가 존재하지 않습니다.");
+            throw new DataAccessException("쿼리 결과가 존재하지 않습니다.");
         }
 
         if (result.size() > 1) {
-            throw new RuntimeException("쿼리 결과가 2개 이상입니다.");
+            throw new DataAccessException("쿼리 결과가 2개 이상입니다.");
         }
 
         return result.get(0);
     }
 
     public void update(String sql, Object... parameters) {
+        Executor executor = (pstmt) -> {
+            pstmt.executeUpdate();
+            return null;
```

Comment:
```suggestion
            return pstmt.executeUpdate();
```
null을 반환하기보다는 executeUpdate값을 반환하는건 어떠신가요?
현재는 update에 반환값이 딱히 없긴하지만 update 쿼리가 동작했을 때 확인하는 용도로 return이 1인지 검증하기도 해요

Reply:
너무 좋네요! 변경하고나니 람다를 사용할 수 있어 코드도 더 깔끔해졌어요~!